runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -11605,6 +11605,149 @@ J9::X86::TreeEvaluator::directCallEvaluator(TR::Node *node, TR::CodeGenerator *c
+   TR::Register *destOffsetReg = cg->gprClobberEvaluate(node->getChild(3), MOV4RegReg);
+   TR::Register *lengthReg = cg->evaluate(node->getChild(4));
+
+   TR::Register *xmmHighReg = cg->allocateRegister(TR_FPR);
+   TR::Register *xmmLowReg = cg->allocateRegister(TR_FPR);
+   TR::Register *zeroReg = cg->allocateRegister(TR_FPR);
+   TR::Register *scratchReg = cg->allocateRegister(TR_GPR);
+
+   TR::RegisterDependencyConditions *deps = generateRegisterDependencyConditions((uint8_t)0, 8, cg);
+
+   deps->addPostCondition(xmmHighReg, TR::RealRegister::NoReg, cg);
+   deps->addPostCondition(xmmLowReg, TR::RealRegister::NoReg, cg);
+   deps->addPostCondition(zeroReg, TR::RealRegister::NoReg, cg);
+   deps->addPostCondition(scratchReg, TR::RealRegister::NoReg, cg);
+   deps->addPostCondition(srcOffsetReg, TR::RealRegister::NoReg, cg);
+   deps->addPostCondition(destOffsetReg, TR::RealRegister::NoReg, cg);
+   deps->stopAddingConditions();
+
+   TR::LabelSymbol *endLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *beginLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *startLoopLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *endLoopLabel = generateLabelSymbol(cg);
+
+   beginLabel->setStartInternalControlFlow();
+   startLoopLabel->setStartInternalControlFlow();
+   endLoopLabel->setEndInternalControlFlow();
+   endLabel->setEndInternalControlFlow();
+
+   generateRegImmInstruction(CMP4RegImm4, node, lengthReg, 0, deps, cg);
+   generateLabelInstruction(JE4, node, endLabel, deps, cg);
+
+   // make sure the register is zero before interleaving
+   generateRegRegInstruction(PXORRegReg, node, zeroReg, zeroReg, deps, cg);
+
+   // vectorized add in loop, 16 bytes per iteration
+   // use srcOffsetReg for loop counter, add starting offset to lengthReg, subtract 16 (xmm register size)
+   // to prevent reading/writing beyond beyond the endLabel of the array
+   generateRegMemInstruction(LEA4RegMem, node, scratchReg, generateX86MemoryReference(lengthReg, srcOffsetReg, 0, -vectorLengthConst, cg), deps, cg);
+
+   generateLabelInstruction(LABEL, node, startLoopLabel, deps, cg);
+   generateRegRegInstruction(CMP4RegReg, node, srcOffsetReg, scratchReg, deps, cg);
+   generateLabelInstruction(JG4, node, endLoopLabel, deps, cg);
+
+   // read 16 bytes into xmmr register
+   generateRegMemInstruction(MOVDQURegMem, node, xmmHighReg, generateX86MemoryReference(srcBufferReg, srcOffsetReg, 0, headerOffsetConst, cg), deps, cg);
+   // TODO; avoid this copy in the future by adding support for VPUNPCK instructions
+   generateRegRegInstruction(MOVDQURegReg, node, xmmLowReg, xmmHighReg, deps, cg);
+   // interleave the low order bytes with zeros and write to the destination buffer
+   generateRegRegInstruction(PUNPCKHBWRegReg, node, xmmLowReg, zeroReg, deps, cg);
+   generateMemRegInstruction(MOVDQUMemReg, node, generateX86MemoryReference(destBufferReg, destOffsetReg, 0, headerOffsetConst + vectorLengthConst, cg), xmmLowReg, deps, cg);
```

Comment:
I see we're adding deps to all instructions in the ICF. This is not functionally wrong, but it will cause the RA to do more work than it needs to. Typically we only add the dependency to the merge label of the ICF.

Reply:
Fixed in [force-push](https://github.com/eclipse-openj9/openj9/compare/54ab43d8c4067d0edf6d01ccfe736c0a6ff981a9..ffcf66373f4530f21659b7b1f6a086145148e5e4).